gfx_scene.h

```diff
@@ -1597,155 +1595,169 @@ class GfxSceneInternal
             if(it != images.end()) material.ao_map = (*it).second;
             materials[&gltf_material] = material_ref;
         }
-        std::map<cgltf_mesh const *, std::vector<GfxConstRef<GfxMesh>>> meshes;
+        typedef std::pair<GfxConstRef<GfxMesh>, GfxConstRef<GfxMaterial>> instance_pair;
+        std::map<cgltf_mesh const *, std::vector<instance_pair>>          meshes;
+        std::map<cgltf_accessor const *, GfxConstRef<GfxMesh>>            meshInstances;
```

Comment:
In GLTF mesh primitives have an index into the buffer region aswell as material and attribute data. Instancing in GLTF occurs when multiple mesh primitives have the same buffer index. So `meshInstances` is introduced to track which buffer indexes (used as a gltf accessor) correspond to which mesh.
We then back up the material used for that primitive as gfx uses that in combination with the transform when making a GfxInstance so it must be stored for later